src/backend/catalog/dependency.c

```diff
@@ -1392,7 +1392,8 @@ deleteOneObject(const ObjectAddress *object, Relation *depRel, int flags)
 	DeleteInitPrivs(object);
 
 	// Delete from ENR - noop if not found from ENR
-	ENRDropEntry(object->objectId);
+	if (object->objectSubId == 0)
```

Comment:
so then how does an ENR entry with `object->objectSubId != 0` get cleaned up?

Reply:
There aren't any ENR entries for such objects - ENR entries are only created for the base table, which would have a subid of 0. This test case is to fix an erroneous case where we are trying to delete something that doesn't actually have an ENR entry at all.

Reply:
nit: please add a comment why we explicitly check for `objectSubId`

Reply:
Sorry, I worded my comment awkwardly - the overall fix requires this change in order to work. Without it, we will get an error when trying to ALTER TABLE DROP COLUMN, as the table will erroneously be deleted when we only wanted to drop one column. So in a sense, all of the test cases for the feature in https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/2753 exercise this code path.

`This test case is to fix an erroneous case` should have been `This check is to fix an erroneous case`.

I'll add a comment clarifying why we need this check.